docs/spectrum1d.rst

```diff
@@ -47,7 +47,22 @@ encouraged to :doc:`create their own loader </custom_loading>`.
     >>> from specutils import Spectrum1D
     >>> spec1d = Spectrum1D.read("/path/to/file.fits")  # doctest: +SKIP
 
+Most of the built-in specutils default loaders can also read an existing 
+`astropy.io.fits.HDUList` object or an open file object (as resulting 
+from e.g. streaming a file from the internet). Note that in these cases, a 
+format string corresponding to an existing loader must be supplied.
 
+.. code-block:: python
+    
+    >>> from astroquery.sdss import SDSS # doctest: +SKIP
+    >>> from specutils import Spectrum1D
+    >>> import urllib
+    >>> specs = SDSS.get_spectra(plate=751, mjd=52251, fiberID=160) # doctest: +SKIP
+    >>> Spectrum1D.read(specs[0], format="SDSS-III/IV spec") # doctest: +SKIP
+    <Spectrum1D(flux=<Quantity [30.596626,...]...>
+    >>> specs = urllib.request.urlopen('https://data.sdss.org/sas/dr14/sdss/spectro/redux/26/spectra/0751/spec-0751-52251-0160.fits') # doctest: +SKIP
+    >>> Spectrum1D.read(specs, format="SDSS-III/IV spec") # doctest: +SKIP
```

Comment:
If you do this instead:

```suggestion
    >>> specs = urllib.request.urlopen('https://data.sdss.org/sas/dr14/sdss/spectro/redux/26/spectra/0751/spec-0751-52251-0160.fits') # doctest: +REMOTE_DATA
    >>> Spectrum1D.read(specs, format="SDSS-III/IV spec") # doctest: +REMOTE_DATA
```

then this part of the doctests will run but only when the --remote-data argument is given, which is good because then this doubles as a unit test (and also a check that the example in the docs still works!)